### ISSUE TITLE
TestQgsGdalProvider::testGdalProviderQuerySublayersFastScan() fails [ netcdf ]

### ISSUE THREAD

**Issue by strk** [qgis/QGIS#48906]
### What is the bug or the crash?

```
FAIL!  : TestQgsGdalProvider::testGdalProviderQuerySublayersFastScan() Compared values are not the same
   Actual   (res.count()): 0
   Expected (1)          : 1
   Loc: [/usr/local/src/qgis/qgis/src/master/tests/src/core/testqgsgdalprovider.cpp(681)]
```

### Steps to reproduce the issue

Run: `xvfb-run ctest --output-on-failure -R test_core_gdalprovider`

### Versions

Current master branch as of June 7th 2022 - e58cb1f02d00cc7256a4e339228cc7dcb0ccd683

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

gdal-config --version returns 3.2.2

See also #48905

**Comment by strk** [comment]
The test passes with GDAL 3.0.4, but I guess it's the available format support, not the gdal version. We'd need a way to SKIP the netcdf test if netcdf support is not exposed by GDAL.

**Comment by strk** [comment]
`gdal-config --formats | grep netcdf` can be used to check if netcdf is available as a format in the installed gdal

**Comment by strk** [comment]
This problem is fixed with the change proposed in #48907 (skip netcdf based tests when driver is unavailable)
